tests/modules/fastani/main.nf

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { FASTANI } from '../../../modules/fastani/main.nf' addParams( options: [:] )
+
+workflow test_fastani {
+
+    query = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    reference = file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
+
+    FASTANI ( [ id:'test' ], query, reference, [], [] )
```

Comment:
I haven't added the test including the `query_list` and `reference_list` mechanism to test multiple fastas as of now.

Reply:
I think it might be worthwhile to get @drpatelh opinion here.

Harshil - FastANI is really good at one-to-one comparisons (`query_fastq`, `reference_fasta`), but when there are thousands of samples, one-to-many (`query_fasta`, `reference_list`) becomes much better in Nextflow. Mostly because instead of thousands of pair-wise outputs, you get one output per-query.

Related issue: https://github.com/bactopia/bactopia/issues/133 

Would you recommend multiple FastANI "modules", one for each type of run?

`fastani query_fasta reference_fasta` (one-to-one)
`fastani query_fasta reference_list` (one-to-many, where `reference_list` is a list of FASTAs)
`fastani query_list reference_fasta` (many-to-one, where `query_list` is a list of FASTAs)
`fastani query_list reference_list` (many-to-many, where `query_list` and `reference_list` are a list of FASTAs)

I think I might have just figured it out though.

I wonder if in this case we make `query` and `reference` an array input. If array length is 1 then treat it like `query_fasta|reference_fasta` otherwise if its >1 create a `query_list|reference_list`

The outputs should be the same regardless of which way FastANI is run.

Reply:
Yup. Exactly. We could treat it a bit like single-end / paired-end data where if the latter then we can access elements within the array within the module.

Reply:
Cool, I have relied on a `batch_input` field to decide upon the branch which should be used 

```
    if (meta.batch_input) {
    """
    fastANI \\
        "-ql $query" \\
        "-rl $reference" \\
        -o ${prefix}.ani.txt

```

Please do let me know if that works